halfsiphash.c

```diff
@@ -69,19 +69,22 @@
 #define TRACE
 #endif
 
-int halfsiphash(const uint8_t *in, const size_t inlen, const uint8_t *k,
-                uint8_t *out, const size_t outlen) {
+int halfsiphash(const void *in, const size_t inlen, const void *k, uint8_t *out,
+                const size_t outlen) {
+
+    unsigned char *ni = (unsigned char *)in;
+    unsigned char *kk = (unsigned char *)k;
```

Comment:
You cast away the `const` qualifier which conflicts with the role of that pointer.

So what about:

```
    const unsigned char *ni = (const unsigned char *)in;
￼    const unsigned char *kk = (const unsigned char *)k;
```

For C compiler only compatibility, you could even leave out the cast on the right hand side.